### PR TITLE
Improve caching of tag name to id mapping #23

### PR DESCRIPTION
**Pull request by mkubala**

### CODE REVIEW

File: core/src/main/scala/akka/persistence/postgres/config/AkkaPersistenceConfig.scala [hunk 1]
```diff
@@ -58,6 +58,7 @@ class SnapshotTableConfiguration(config: Config) {
 }
 
 class JournalPluginConfig(config: Config) {
+  // TODO get rid of tagSeparator (incl docs)
```
Comment (devcateu) [inline core/src/main/scala/akka/persistence/postgres/config/AkkaPersistenceConfig.scala:61]: todos

File: core/src/main/scala/akka/persistence/postgres/journal/dao/ByteArrayJournalDao.scala [hunk 2]
```diff
@@ -92,35 +91,24 @@ trait BaseByteArrayJournalDao extends JournalDaoWithUpdates with BaseJournalDaoW
...
+  def asyncWriteMessages(messages: Seq[AtomicWrite]): Future[Seq[Try[Unit]]] =
+    Future
+      .sequence {
+        serializer
+          .serialize(messages)
+          // If serialization fails for some AtomicWrites, the other AtomicWrites may still be written
+          .map(_.map(Success(_)).recover {
+            case ex =>
+              Failure(ex)
+          })
```
Comment (devcateu) [inline core/src/main/scala/akka/persistence/postgres/journal/dao/ByteArrayJournalDao.scala:103]: Maybe put it in serialize method?

File: core/src/test/resources/schema/postgres/partitioned-schema.sql [hunk 3]
```diff
@@ -36,6 +36,7 @@ CREATE TABLE IF NOT EXISTS public.journal
 ) PARTITION BY LIST (persistence_id);
 
 CREATE INDEX journal_ordering_idx ON public.journal USING BRIN (ordering);
+CREATE INDEX journal_tags_idx ON public.journal USING GIN(tags);
```
Comment (devcateu) [inline core/src/test/resources/schema/postgres/partitioned-schema.sql:39]: Do you run perf-tests? From my previous tests it look like GIN index does not improve performance in our case.

Comment (mkubala) [reply core/src/test/resources/schema/postgres/partitioned-schema.sql:39]: I've been comparing execution plans (query costs)

File: core/src/main/scala/akka/persistence/postgres/journal/dao/ByteArrayJournalDao.scala (hunk 2 above)
Comment (mkubala) [reply core/src/main/scala/akka/persistence/postgres/journal/dao/ByteArrayJournalDao.scala:103]: I'm not sure. Wrapping failed serialization in a `scala.util.Failure` is used there in order to allow JournalDao persist the rest of events that have been successfuly encoded. Change of DAO's behaviour should not affect the serializer component.

**Pull request merged**